### PR TITLE
Fix Issue 17467 - BitArray are broken with <<= 64

### PR DESCRIPTION
**Pull request by Darredevil**

### CODE REVIEW

**Comment by Darredevil** [comment]
PS. Fixed rollRight as well.

**Comment by RazvanN7** [comment]
LGTM.

File: std/bitmanip.d [hunk 1]
```diff
@@ -1979,7 +1979,10 @@ public:
     }
     body
     {
-        return (upper << (bitsPerSizeT - nbits)) | (lower >> nbits);
+        if (nbits == 0)
+            return lower;
+        else
```
Comment (andralex) [inline std/bitmanip.d]: remove

File: std/bitmanip.d [hunk 2]
```diff
@@ -2014,7 +2017,10 @@ public:
     }
     body
     {
-        return (upper << nbits) | (lower >> (bitsPerSizeT - nbits));
+        if (nbits == 0)
+            return upper;
+        else
```
Comment (andralex) [inline std/bitmanip.d]: remove

Comment (Darredevil) [reply std/bitmanip.d]: Fixed.

**Pull request merged**